homeassistant/components/switch/mfi.py

```diff
@@ -0,0 +1,122 @@
+"""
+homeassistant.components.switch.mfi
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Support for Ubiquiti mFi Switches.
+
+Configuration:
+
+sensor:
+  platform: mfi
+  host: ADDRESS
+  port: PORT
+  username: USERNAME
+  password: PASSWORD
+
+Variable:
+
+host
+*Required
+ADDRESS is the IP or hostname of your mFi controller.
+
+port
+*Optional
+PORT is the port of your mFi controller (usually 6443)
+
+username
+*Required
+USERNAME is the mFi admin username
+
+password
+*Required
+PASSWORD is the mFi admin user's password
+"""
+
+import logging
+
+from homeassistant.components.switch import DOMAIN, SwitchDevice
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.const import STATE_ON, STATE_OFF
+from homeassistant.helpers import validate_config
+
+REQUIREMENTS = ['mficlient==0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+SWITCH_MODELS = [
+    'Outlet',
+]
+
+
+# pylint: disable=unused-variable
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up mFi sensors. """
+
+    if not validate_config({DOMAIN: config},
+                           {DOMAIN: ['host',
+                                     CONF_USERNAME,
+                                     CONF_PASSWORD]},
+                           _LOGGER):
+        _LOGGER.error('A host, username, and password are required')
+        return False
+
+    host = config.get('host')
+    port = int(config.get('port', 6443))
+    username = config.get('username')
+    password = config.get('password')
+
+    from mficlient.client import MFiClient
+
+    try:
+        client = MFiClient(host, username, password, port=port)
+    except client.FailedToLogin as ex:
+        _LOGGER.error('Unable to connect to mFi: %s', str(ex))
+        return False
+
+    for device in client.get_devices():
+        for port in device.ports.values():
+            if port.model in SWITCH_MODELS:
+                add_devices([MfiSwitch(port)])
+
+
+class MfiSwitch(SwitchDevice):
+    """ An mFi switch-able device. """
+    def __init__(self, port):
+        self._port = port
+
+    @property
+    def should_poll(self):
+        return True
+
+    @property
+    def unique_id(self):
+        return self._port.ident
+
+    @property
+    def name(self):
+        return self._port.label
+
+    @property
+    def state(self):
```

Comment:
Instead of implementing `state` you can implement `is_on` and return `self._port.output`